src/diffusers/loaders/single_file_utils.py

```diff
@@ -1808,4 +1808,17 @@ def create_diffusers_t5_model_from_checkpoint(
 
     else:
         model.load_state_dict(diffusers_format_checkpoint)
+
```

Comment:
ohh is this related to this? 
https://github.com/huggingface/diffusers/issues/8604#issuecomment-2174675284

Reply:
Yeah exactly. When you cast the entire model to `fp16` the images will be corrupted. You have to use the logic in `from_pretrained`.